uacp/conn.go

```diff
@@ -134,6 +135,21 @@ func Listen(ctx context.Context, endpoint string, ack *Acknowledge) (*Listener,
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := strings.CutSuffix(laddr.Host, ":0"); ok {
```

Comment:
I'd prefer if you could extract this code into a separate function and write a table driven test for that.

Reply:
refactored in https://github.com/gopcua/opcua/pull/798/commits/91c151828d1f116ee1177f17f0e187ef062b9e21

What kind of input did you have in mind for the table-driven test? (guidance appreciated)